### PR TITLE
Simplify manifest

### PR DESCRIPTION
**Pull request by Anilm3**
Simplify the use of the manifest:
- Reduce string operations.
- Automatically determine the parent / child relationship between addresses.
- Add support for `key_path.size() > 1`.
- Introduce manifest builder to avoid storing unnecessary information.

### CODE REVIEW

File: src/manifest.hpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+protected:
+    std::unordered_map<std::string, target_type> targets_{};
+    std::unordered_map<target_type, target_info> info_{};
+
+    // Root address memory to be returned to the API caller
+    std::vector<const char*> root_addresses_;
+};
+
+class manifest_builder {
```
Comment (estringana) [inline src/manifest.hpp:96]: Out of interest. Is it ok to have multiple classes per hpp file on C++?

**Review by estringana — APPROVED** [review]

File: src/manifest.hpp (hunk 1 above)
Comment (Anilm3) [reply src/manifest.hpp:96]: Yes, although ideally they should be related somehow.

**Review by cataphract — APPROVED** [review]

File: src/manifest.hpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            hash ^= std::hash<std::string>{}(str);
+        }
+        return hash;
+    }
+};
+}
+
+namespace ddwaf
+{
+class manifest {
```
[the remark this answers is not in the document]
Comment (Anilm3) [reply src/manifest.hpp:35]: My objective was to remove it altogether but it's fine as a simple abstraction for now.

**Comment by Anilm3** [comment]
I'll address any comments on a future PR since none of them are critical.

**Pull request merged**